### PR TITLE
Added customizable margins and cleaned the dimension code

### LINKED ISSUE #223: Wrong number rendering on yAxis
Hi, 
Great job with that lib but I found some glitch with rendering numbers on yAxis, here is plunker that shows that problem [http://plnkr.co/edit/zwTeO1HOwqri4f2hgNrg?p=preview](http://plnkr.co/edit/zwTeO1HOwqri4f2hgNrg?p=preview). (edit) Problem shows when you are setting number greater equal then 100000

Problem detected on google-chrome 42.0.2311.135 and firefox 37.0.2.

Is it possible to solve that or do you have some quick workaround?

Br,
Bart

### PR DESCRIPTION
**Pull request by chaosmail**
Hi n3-charts team,

I added customizable margins. To do so, I needed to clean the code dealing with dimensions - I think it is a little cleaner now. I added a demo here [http://plnkr.co/edit/naTVhb6JJrSKm9BL3Rg5](http://plnkr.co/edit/naTVhb6JJrSKm9BL3Rg5?p=preview).

It can be used like this:
```js
$scope.options = {
  ...
  margin: {
    left: 100
  }
}
```

Fixes #223

Best,
Christoph

### CODE REVIEW

File: test/unit/options.mocha.coffee [hunk 1]
```diff
@@ -78,6 +78,27 @@ describe 'options', ->
...
+      expect(o.margin.left).to.equal 50
+
+    it 'should use the default thumbnail margin', ->
+      o = n3utils.sanitizeOptions(null, 'thumbnail')
+      expect(o.margin.top).to.equal 1
+      expect(o.margin.right).to.equal 1
+      expect(o.margin.bottom).to.equal 2
+      expect(o.margin.left).to.equal 0
+
+    it 'should use parse margins as float', ->
```
Comment (lorem--ipsum) [inline test/unit/options.mocha.coffee]: Typo here I guess ("use" is of no use ^^)

File: test/unit/size.mocha.coffee [hunk 2]
```diff
@@ -16,23 +16,30 @@ describe 'size', ->
...
+    {element, outerScope} = pepito.directive("""
+      <div>
+        <linechart data='data' options='options'></linechart>
+      </div>
+      """,
+      (element) ->
+        innerScope = element.children()[0].aElement.isolateScope()
+      )
+
+    innerScope = element.childByClass('chart').aElement.isolateScope()
```
Comment (lorem--ipsum) [inline test/unit/size.mocha.coffee]: How comes you reassign `innerScope` ? Is that necessary ?

File: test/unit/testUtils.coffee [hunk 3]
```diff
@@ -72,6 +72,20 @@ utils.factory 'pepito', ($compile, $rootScope, fakeMouse) ->
...
+  e = document.createEvent('UIEvent')
+
+  # We could add cool stuff here, e.g resize
+  # to a dimension 800x400
+  resize = () ->
+    e.initEvent 'resize', true, false
+    $window.dispatchEvent e
+
+  return {
+    resize: () -> resize(arguments)
```
Comment (lorem--ipsum) [inline test/unit/testUtils.coffee]: I think you can just write
```js
return {
  resize
}
```

(just saying, I'll merge this anyway)

**Comment by lorem--ipsum** [comment]
It looks definitely cleaner ! Please rebase and I'll merge it. I think the merge conflicts are due to the fact that the build code is versioned...

**Comment by chaosmail** [comment]
Here it is, I addressed all your remarks ;)

**Comment by lorem--ipsum** [comment]
Sweet !

**Comment by lorem--ipsum** [comment]
:poultry_leg:

**Comment by chaosmail** [comment]
Fixes #171 
however, the issue needs to be closed manually

**Pull request merged**